clinica/utils/inputs.py

```diff
@@ -0,0 +1,223 @@
+
+        Note:
+            This function is case insensitive, meaning that the pattern argument can, for example, contain maj letter
+            that do not exists in the existing file path.
+
+    """
+
+    from os.path import join, isdir
+    from os import listdir
+    from clinica.utils.io import check_bids_folder, check_caps_folder
+    from colorama import Fore
+
+    def determine_caps_or_bids(input_dir):
+        """
+        Determines if the input is a CAPS or a BIDS folder
+        :param input_dir: input folder
+        :return: True if input_dir is a bids, False if input_dir is a CAPS
+
+            raise :
+                RuntimeError if function could not determine if BIDS or CAPS or whatever else
+        """
+        if isdir(join(input_directory, 'subjects')):
+            if len([sub for sub in listdir(join(input_dir, 'subjects')) if (sub.startswith('sub-') and isdir(join(input_dir, 'subjects', sub)))]) > 0 or isdir(join(input_dir, 'groups')):
+                return False
+            else:
+                raise RuntimeError('Could not determine if ' + input_dir + ' is a CAPS or BIDS directory')
+
+        else:
+            if len([sub for sub in listdir(input_dir) if (sub.startswith('sub-') and isdir(join(input_dir, sub)))]) > 0:
+                return True
+            else:
+                if isdir(join(input_dir, 'groups')):
+                    return False
+                else:
+                    raise RuntimeError('Could not determine if ' + input_dir + ' is a CAPS or BIDS directory')
+
+    assert isinstance(information, dict), 'A dict must be provided for the argmuent \'dict\''
+    assert all(elem in information.keys() for elem in ['pattern', 'description']), '\'information\' must contain the keys \'pattern\' and \'description'
+    assert all(elem in ['pattern', 'description', 'needed_pipeline'] for elem in information.keys()), '\'information\' can only contain the keys \'pattern\', \'description\' and \'needed_pipeline\''
+
+    pattern = information['pattern']
+    is_bids = determine_caps_or_bids(input_directory)
+
+    if is_bids:
+        check_bids_folder(input_directory)
+    else:
+        check_caps_folder(input_directory)
+
+    # Some check on the formatting on the data
+    assert pattern[0] != '/', 'pattern argument cannot start with char : / (does not work in os.path.join function). ' \
+                              + 'If you want to indicate the exact name of the file, use the format' \
+                              + ' directory_name/filename.extension or filename.extensionin the pattern argument'
+    assert len(subjects) == len(sessions), 'Subjects and sessions must have the same length'
+    if len(subjects) == 0:
+        raise RuntimeError('No subjects and sessions provided.')
+
+    # rez is the list containing the results
+    rez = []
```

Comment:
Could be more readable:
```suggestion
    results = []
```

Reply:
Done